auth/jwt_bearer.py

```diff
@@ -0,0 +1,59 @@
+# Code copied from github.com/overrideveloper/HowAreYouApi*
+
+from bson import ObjectId
+from fastapi.security import HTTPBearer, HTTPAuthorizationCredentials
+from fastapi import Request, HTTPException
+
+from helper.helpers import user_helper
+from models.Database import Database
+from models.model import TokenPayload
+from auth.encode_decode import decodeJWT
+from datetime import time, date
+from typing import Dict
+import time
+
+class JWTBearer(HTTPBearer):
+    db: Database = None
+    def __init__(self, db, auto_error: bool = True):
+        super(JWTBearer, self).__init__(auto_error=auto_error)
+        self.db = db
+
+    async def __call__(self, request: Request):
+        credentials: HTTPAuthorizationCredentials = await super(JWTBearer, self).__call__(request)
+
+        if credentials:
+            if not credentials.scheme == "Bearer":
+                print("Failed here.")
+                raise HTTPException(status_code=409, detail="You are not authorized to access this resource")
+
+            if not self.verify_jwt(credentials.credentials):
+                print("Failed here two")
+                raise HTTPException(status_code=401, detail="You are not authorized to access this resource")
+
+            return credentials.credentials
+        else:
+            print("Failed here three")
+            raise HTTPException(status_code=401, detail="You are not authorized to access this resource")
+
+    def verify_jwt(self, jwtoken: str) -> bool:
+        payload: TokenPayload = None
+        isTokenValid: bool = False
+
+        try:
+            payload = TokenPayload(**decodeJWT(jwtoken))
+        except:
+            payload = None
+
+        if payload:
+            if time.mktime(date.today().timetuple()) >= payload.expires:
```

Comment:
I would move the expiry check into the `decodeJWT` method, that should be the responsibility of that method.

I would also move the code that sets the expiry value into the `encodeJWT` method, that should be handle by the method.

Reply:
Okay. So I move all these validation statements to the `decode` & `encode` methods to reduce the story here? Thanks

Reply:
Yeah.